src/index.ts

```diff
@@ -187,9 +187,9 @@ export function basicAuth(userOptions: Partial<BasicAuthOptions> = {}) {
     app
       .state('basicAuthRealm', null as string | null)
       .state('basicAuthUser', null as string | null)
-      .addError({ BASIC_AUTH_ERROR: BasicAuthError })
-      .onError(({ code, error }) => {
-        if (code === 'BASIC_AUTH_ERROR' && error.realm === options.realm) {
+      .error({ BASIC_AUTH_ERROR: BasicAuthError })
```

Comment:
It's possible it's not worth leaving this here since it's not working anymore — wondering if you have any preferences or ideas though? I have not filed a bug upstream to Elysia but there probably should be one.

Reply:
@nathankleyn  please see https://github.com/eelkevdbos/elysia-basic-auth/pull/3  Elysia is looking for a property called `code` inside the error object.